libraries/SRV_Channel/SRV_Channel.cpp

```diff
@@ -194,6 +194,26 @@ uint16_t SRV_Channel::pwm_from_range(float scaled_value) const
     return servo_min + uint16_t( (scaled_value * (float)(servo_max - servo_min)) / (float)high_out );
 }
 
+// convert a pwm to a 0..1 value
+float SRV_Channel::range_from_pwm() const
+{
+    if (servo_max <= servo_min || high_out == 0) {
+        return 0;
+    }
+    if (output_pwm < servo_min) {
+        return 0;
+    }
+    if (output_pwm > servo_max) {
+        return 1;
+    }
+    float full_range = servo_max - servo_min;
+    float value = (float)(output_pwm - servo_min) / full_range;
```

Comment:
In fact I think you could just use `linear_interpolate(0.0, 1.0, output_pwm, servo_min, servo_max)`